### PR TITLE
doc: sync github to spaces

### PR DESCRIPTION
**Pull request by borisdayma**
Explains how to sync a Github repo with Spaces

### CODE REVIEW

File: docs/hub/spaces.md [hunk 1]
```diff
@@ -78,16 +78,70 @@ Those secrets will be exposed to your app using the [Streamlit Secrets](https://
...
+
 - `st.experimental_get_query_params()` and `st.experimental_set_query_params(**parameter)` to manage app state in the url
 - if something doesn't work, please reach out.
 
+## How can I manage my app through Github
+
+Github is great for collaboration. You can keep your app in sync with your Github repository by leveraging Github Actions:
+
+- We require Git LFS for files above 10MB so you may need to review your files if you don't want to use Git LFS. This includes your history. You can use handy tools such as [BFG Repo-Cleaner](https://rtyley.github.io/bfg-repo-cleaner/) to remove the large files from your history (keep a local copy of your repository for backup).
+- Set your Github repository and your Spaces app initially in sync (you may need to force push).
```
Comment (Pierrci) [inline docs/hub/spaces.md:90]:
```suggestion
- Set your Github repository and your Spaces app initially in sync: to add your Spaces app as an additional remote to your existing git repository, you can use the command `git remote add space https://huggingface.co/spaces/FULL_SPACE_NAME`. You can then force-push to sync everything for the first time: `git push --force space main`
```

File: docs/hub/spaces.md [hunk 2]
```diff
@@ -78,16 +78,70 @@ Those secrets will be exposed to your app using the [Streamlit Secrets](https://
...
+  sync-to-hub:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Push to hub
+        env:
+          HF_TOKEN: ${{ secrets.HF_TOKEN }}
+        run: git push https://HF_USERNAME:$HF_TOKEN@huggingface.co/spaces/flax-community/dalle-mini main
```
Comment (Pierrci) [inline docs/hub/spaces.md:113]:
```suggestion
        run: git push https://HF_USERNAME:$HF_TOKEN@huggingface.co/spaces/FULL_SPACE_NAME main
```

File: docs/hub/spaces.md [hunk 3]
```diff
@@ -78,16 +78,70 @@ Those secrets will be exposed to your app using the [Streamlit Secrets](https://
...
 - `st.experimental_get_query_params()` and `st.experimental_set_query_params(**parameter)` to manage app state in the url
 - if something doesn't work, please reach out.
 
+## How can I manage my app through Github
+
+Github is great for collaboration. You can keep your app in sync with your Github repository by leveraging Github Actions:
+
+- We require Git LFS for files above 10MB so you may need to review your files if you don't want to use Git LFS. This includes your history. You can use handy tools such as [BFG Repo-Cleaner](https://rtyley.github.io/bfg-repo-cleaner/) to remove the large files from your history (keep a local copy of your repository for backup).
+- Set your Github repository and your Spaces app initially in sync (you may need to force push).
+- Set up a Github Action to push your Github main branch automatically to Spaces: replace `HF_USERNAME` with your Hugging Face username and [create a Github secret](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-an-environment) `HF_TOKEN` containing your Hugging Face API token.
```
Comment (Pierrci) [inline docs/hub/spaces.md:91]:
```suggestion
- Set up a Github Action to push your Github main branch automatically to Spaces: replace `HF_USERNAME` with your Hugging Face username, `FULL_SPACE_NAME` with your Spaces name, and [create a Github secret](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-an-environment) `HF_TOKEN` containing your Hugging Face API token.
```

File: docs/hub/spaces.md [hunk 4]
```diff
@@ -78,16 +78,70 @@ Those secrets will be exposed to your app using the [Streamlit Secrets](https://
...
+          HF_TOKEN: ${{ secrets.HF_TOKEN }}
+        run: git push https://HF_USERNAME:$HF_TOKEN@huggingface.co/spaces/flax-community/dalle-mini main
+```
+
+- Create an action so file sizes are automatically checked on any new PR
+
+```
+name: Check file size
+
+on:
```
Comment (Pierrci) [inline docs/hub/spaces.md:121]:
```suggestion
on:               # or directly `on: [push]` to run the action on every push on any branch
```

**Review by Pierrci — COMMENTED** [review]
Thank you very much @borisdayma, I added a few suggestions :)

File: docs/hub/spaces.md [hunk 5]
```diff
@@ -78,16 +78,70 @@ Those secrets will be exposed to your app using the [Streamlit Secrets](https://
...
+  workflow_dispatch:
+
+jobs:
+  sync-to-hub:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check large files
+        uses: ActionsDesk/lfs-warning@v2.0
+        with:
+          filesizelimit: 9000000 # so we can sync to HF spaces
```
Comment (julien-c) [inline docs/hub/spaces.md:135]:
```suggestion
          filesizelimit: 10485760 # = 10MB, so we can sync to HF spaces
```

Let's use the exact same value as in hf.co, no?

Comment (julien-c) [reply docs/hub/spaces.md]: (cc @Pierrci)

**Comment by julien-c** [comment]
Thanks @borisdayma!

**Comment by julien-c** [comment]
deployed: https://huggingface.co/docs/hub/spaces#how-can-i-manage-my-app-through-github (just added syntax coloring for the yaml)

**Comment by julien-c** [comment]
You should share this and we'll amplify :)

**Comment by borisdayma** [comment]
btw @Pierrci, not sure about the `on: [push]` working on forks (but maybe it does)

**Comment by julien-c** [comment]
it does, but we (repo admins) need to enable it to run the first time from any given (external) user's fork

**Comment by julien-c** [comment]
I didn't run it here b/c it's only doc

**Comment by julien-c** [comment]
Updated in https://github.com/huggingface/hub-docs/pull/436

**Pull request merged**